### PR TITLE
DeNaN: instrument parameters as they arrive in each function

### PR DESCRIPTION
**Pull request by kripken**
Instead of instrumenting every `local.get`, instrument parameters
on arrival at a function once on entry. After that, every local will
always contain a de-naned value (since we would denan on a
`local.set`). This is more efficient and also less confusing I think.

### CODE REVIEW

File: src/passes/DeNaN.cpp [hunk 1]
```diff
@@ -61,6 +67,33 @@ struct DeNaN : public WalkerPass<
...
+      } else if (func->getLocalType(i) == Type::f64) {
+        fixes.push_back(builder.makeLocalSet(
+          i,
+          builder.makeCall(
+            "deNan64", {builder.makeLocalGet(i, Type::f64)}, Type::f64)));
+      }
+    }
+    if (!fixes.empty()) {
+      fixes.push_back(func->body);
+      func->body = builder.makeBlock(fixes);
```
[the remark this answers is not in the document]
Comment (kripken) [reply src/passes/DeNaN.cpp:93]: We could run `--merge-blocks` perhaps to do that.

**Comment by kripken** [comment]
I don't think we can exclude `local.tee` in general because it sets a new value, but maybe we can look into optimizing that pattern specifically.

**Comment by aheejin** [comment]
Why not? Maybe I'm misunderstanding something... The expressions that had been already stored in the locals were deNaNed when they were stored, so we don't need to deNaN them when we load it again from it..?

**Comment by kripken** [comment]
Oh sorry, I misunderstood. Yeah, since `tee` just lets a value fall through, without changing it, we can ignore it, very good point.

Added that now, and also added merge-blocks.

**Pull request merged**